doc/ip/revoker.md

```diff
@@ -0,0 +1,15 @@
+# CHERIoT hardware revocation engine
+
+This block is integrated from the [CHERIoT Safe repository](https://github.com/microsoft/cheriot-safe/blob/main/src/msft_cheri_subsystem/msftDvIp_mmreg.sv).
+It essentially sweeps memory for revoked capabilities.
+For more information on how this works please refer to [Section 3.3.3 (background pipelined revoker) of this paper](https://cheriot.org/papers/2023-micro-cheriot-uarch.pdf).
+
+These are the register offsets:
+
+| Offset | Register |
+|--------|----------|
+| 0x0000 | Start address to begin sweeping
+| 0x0004 | End address to stop sweeping
+| 0x0008 | Set when sweeping is happening (most significant byte is set to constant 0x55)
+| 0x0012 | Set when sweeping is done
+| 0x0014 | Enable bit for raising an interrupt when sweeping is done
```

Comment:
These don't seem quite right:

0x08 is the control word.  The high bits are 0x55, writing the low bit triggers the revoker to start, reading is unspecified.

0x0c is the epoch word.  It should be a monotonic (wrapping) counter that is incremented when the revocation starts and when it completes (so the low bit is 1 during a revocation sweep).

0x10 is the interrupt status word.  We never actually use this in software.

0x12 is not a 32-bit aligned address.

0x14 is the interrupt-enable word.  This is used to request an interrupt when the current revocation sweep ends.

Reply:
Thanks, I've updated this now.